### DISCUSSION TITLE
[Feature]: Quality selector

### DISCUSSION THREAD

**Discussion by sedatb** [TheWidlarzGroup/react-native-video#4980] [Ideas]
### What problem are you trying to solve?

The user wants to be able to select the video quality when playing media with this package.

### Proposed solution

Add a quality selector that lets the consumer choose the playback quality.

### Additional context

/cc @KrzysztofMoch

**Comment by moskalakamil** [comment]
Hi! Thanks for posting feature idea - actually this is not on our current roadmap but if you need this [please contact us](https://www.thewidlarzgroup.com#Contact) - we can offer you paid support for this feature

**Comment by sgebr01** [comment]
I'm using `selectedVideoTrack` on android for this (I don't believe iOS supports this natively for manifest-based streaming anyway).

**Comment by sedatb** [comment]
@sgebr01 we use selectedVideoTrack type "auto" on android but, It cannot automatically switch from high quality to low quality. The video freezes.

**Comment by gustavoabel** [comment]
> [@sgebr01](https://github.com/sgebr01) we use selectedVideoTrack type "auto" on android but, It cannot automatically switch from high quality to low quality. The video freezes.

same here

**Comment by TWG-AI** [comment]
Thanks for the idea — noted for the team to consider.

> I'm an AI assistant, so double-check any specifics I may have mentioned.

**Comment by TWG-AI** [comment]
This request overlaps significantly with discussion [#4962](https://github.com/TheWidlarzGroup/react-native-video/discussions/4962), which also covers manual quality selection for HLS streams. Tracking both together.

> I'm an AI assistant — double-check any details above.
